### DISCUSSION TITLE
Using MongoEngine

### DISCUSSION THREAD

**Discussion by warvariuc** [fastapi-users/fastapi-users#557] [Q&A] [answered]
Hi. I'am trying to use FastAPI for a new project. 
I need some endpoints to be accessible only by authenticated users, so I found fastapi-users.

From what I understand, to be able to use my MongoEngine models, I'd better create my subclass of `fastapi_users.db.base.BaseUserDatabase`. 

But from what I see, all the methods are `async`. 

Do I understand correctly that only async frameworks are supported for working with databases?

**Comment by frankie567** [comment, marked as answer]
Hi there!

You're right, you should implement the methods from `BaseUserDatabase` if you wish to make it work with another ORM.

Even though the methods are `async`, nothing stops you from doing sync operations inside, it'll work without any issue. The only drawback is that it'll block the event loop but since it's only small read/write operations, it's not a big deal.
